packages/rum/src/rumSession.ts

```diff
@@ -15,19 +15,53 @@ export enum RumSessionType {
   TRACKED_WITHOUT_RESOURCES = '2',
 }
 
+class StoredSession {
+  private id: string | undefined
+  private type: RumSessionType | undefined
+
+  constructor(private isAlive: () => boolean) {}
+
+  store(id: string | undefined, type: RumSessionType | undefined) {
+    this.id = id
+    this.type = type
+  }
+
+  getId() {
+    this.makeSureSessionIsAlive()
+    return this.id
+  }
+
+  isTracked() {
+    this.makeSureSessionIsAlive()
+    return isTracked(this.type)
+  }
+
+  isTrackedWithResource() {
+    this.makeSureSessionIsAlive()
+    return this.type === RumSessionType.TRACKED_WITH_RESOURCES
+  }
+
+  private makeSureSessionIsAlive() {
+    if (!this.isAlive()) {
+      this.id = undefined
+      this.type = undefined
+    }
+  }
+}
+
 export function startRumSession(configuration: Configuration, lifeCycle: LifeCycle): RumSession {
   const session = startSessionManagement(RUM_SESSION_KEY, (rawType) => computeSessionState(configuration, rawType))
+  const storedSession = new StoredSession(() => session.getId() !== undefined)
+
+  storedSession.store(session.getId(), session.getType())
 
   session.renewObservable.subscribe(() => {
+    lifeCycle.notify(LifeCycleEventType.SESSION_WILL_RENEW)
+    storedSession.store(session.getId(), session.getType())
     lifeCycle.notify(LifeCycleEventType.SESSION_RENEWED)
   })
```

Comment:
Since the currentSessionId is already saved in session management and since we already have an object for session, why not only add something like:
```ts 
session.beforeRenewObservable.subscribe(() => {
  lifeCycle.notify(LifeCycleEventType.BEFORE_SESSION_RENEWED)
})
```

Reply:
This was my first thought, but I found that it's not that simple.

We can move the logic to sessionManagement, but because `session.getId()` is directly accessing the cookie value, we have no guarantee that the id right before a renew event is the one from the previously active session. It can already by the new session id or another session id. We'd still need to get rid of those direct cookie accesses, with a similar logic.

WDYT? Should I move this logic into sessionManagement?

Reply:
FMU, there is two use cases:
- either we want the active session
- either we want the cached session
Could we expose from sessionManagement:
```ts
getCachedId() {
  return currentSessionId
}
```
And use that where we used viewContext.sessionId?

Reply:
This can be a good solution, yes. But we have to make sure that the session is still alive before using the cached id, else [this test](https://github.com/DataDog/browser-sdk/blob/master/test/e2e/scenario/agents.scenario.ts#L165-L171) fails.

Where would we want the active session outside of the `sessionManagement` module?

Reply:
> This can be a good solution, yes. But we have to make sure that the session is still alive before using the cached id, else this test fails.

this rumSession check should still work, right?
```ts
isTracked: () => session.getId() !== undefined && isTracked(session.getType()),
```

> Also, where would we want the active session outside of the sessionManagement module?

in rumSession and loggerSession then.
Maybe using only cached session for rumSession.getId could work. renaming it to getCachedId could make this behavior more explicit

Reply:
> > This can be a good solution, yes. But we have to make sure that the session is still alive before using the cached id, else this test fails.
> 
> this rumSession check should still work, right?
> 
> ```ts
> isTracked: () => session.getId() !== undefined && isTracked(session.getType()),
> ```
IMO, the session type should be cached alongside the id, else we'll have incoherences.
> 
> > Also, where would we want the active session outside of the sessionManagement module?
> 
> in rumSession and loggerSession then.
> Maybe using only cached session for rumSession.getId could work. renaming it to getCachedId could make this behavior more explicit
* Like you say, in rumSession we could use only cached session informations.
* IMO, in loggerSession, nothing prevents us to use cached session informations as well. In fact, it should already use the session id from RUM since it is exposed in internal context
* Thus, having a single function to retrieve session information would make sense to me (whether this information is cached or not).

Reply:
> IMO, the session type should be cached alongside the id, else we'll have incoherences.

session type is only computed once at startRum but we could move that to rumSession if it makes more sense

> Like you say, in rumSession we could use only cached session informations.

What I had in mind was return cached value for getId, use cookie value for isTracked/isTrackedWithResource

> IMO, in loggerSession, nothing prevents us to use cached session informations as well. In fact, it should already use the session id from RUM since it is exposed in internal context

I disagree, there is no reason to use cached value for logs since we don't keep local state like view for logs.
Since we need to compute session id when rum is not present, why should we change the current behavior?

> Thus, having a single function to retrieve session information would make sense to me (whether this information is cached or not).

not sure to see what you have in mind here